src/Compilers/VisualBasic/Test/Semantic/Binding/LookupTests.vb

```diff
@@ -2800,6 +2800,237 @@ BC37229: 'C' is ambiguous between declarations in namespaces 'A.X, B.X'.
+                                                    ]]></expected>)
+        End Sub
+
+        <Fact, WorkItem(2909, "https://github.com/dotnet/roslyn/issues/2909")>
+        Public Sub AmbiguousEnumConstants02c()
+            Dim csCompilation = CreateCSharpCompilation("CSEnum",
+            <![CDATA[
+public enum Color
+{
+Red,
+Green,
+DateTime,
+[System.Obsolete] Datetime = DateTime,
+DATETIME,
+Blue,
+}
+]]>,
+                compilationOptions:=New Microsoft.CodeAnalysis.CSharp.CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary))
+            csCompilation.VerifyDiagnostics()
+            Dim vbCompilation = CreateVisualBasicCompilation("VBEnumClient",
+            <![CDATA[
+Public Module Program
+    Sub Main()
+        System.Console.WriteLine(CInt(Color.DateTime))
+    End Sub
+End Module]]>,
+                compilationOptions:=New VisualBasicCompilationOptions(OutputKind.ConsoleApplication),
+                referencedCompilations:={csCompilation})
+            CompilationUtils.AssertTheseDiagnostics(vbCompilation,
+                                                    <expected><![CDATA[
+BC31429: 'DateTime' is ambiguous because multiple kinds of members with this name exist in enum 'Color'.
+        System.Console.WriteLine(CInt(Color.DateTime))
+                                      ~~~~~~~~~~~~~~
+                                                    ]]></expected>)
+        End Sub
+
+        <Fact, WorkItem(2909, "https://github.com/dotnet/roslyn/issues/2909")>
+        Public Sub AmbiguousEnumConstants02d()
+            Dim vbCompilation1 = CreateVisualBasicCompilation("VBEnum",
+            <![CDATA[
+Public Enum Color
+    Red
+    Green
+    DateTime
+    <System.Obsolete> Datetime = DateTime
+    DATETIME
+    Blue
+End Enum
+]]>,
+                compilationOptions:=New VisualBasicCompilationOptions(OutputKind.DynamicallyLinkedLibrary))
+            CompilationUtils.AssertTheseDiagnostics(vbCompilation1,
+                                                    <expected><![CDATA[
+BC31421: 'Datetime' is already declared in this enum.
+    <System.Obsolete> Datetime = DateTime
+                      ~~~~~~~~
+BC31429: 'DateTime' is ambiguous because multiple kinds of members with this name exist in enum 'Color'.
+    <System.Obsolete> Datetime = DateTime
+                                 ~~~~~~~~
+BC31421: 'DATETIME' is already declared in this enum.
+    DATETIME
+    ~~~~~~~~
+                                                    ]]></expected>)
+            Dim vbCompilation = CreateVisualBasicCompilation("VBEnumClient",
+            <![CDATA[
+Public Module Program
+    Sub Main()
+        System.Console.WriteLine(CInt(Color.DateTime))
+    End Sub
+End Module]]>,
+                compilationOptions:=New VisualBasicCompilationOptions(OutputKind.ConsoleApplication),
+                referencedAssemblies:={New VisualBasicCompilationReference(vbCompilation1), MscorlibRef, MsvbRef})
```

Comment:
Since we are explicitly giving the list of referenced assemblies, we do not get to take advantage of the default list provided by this helper function when no list of referenced assemblies is provided.